nilearn/plotting/surf_plotting.py

```diff
@@ -3,18 +3,440 @@
+    all_samples = []
+    for image in images:
+        interpolator = interpolate.RegularGridInterpolator(
+            grid, image,
+            bounds_error=False, method='linear', fill_value=None)
+        samples = interpolator(interp_locations)
+        samples[masked] = np.nan
+        all_samples.append(samples)
+    all_samples = np.asarray(all_samples)
+    all_samples = all_samples.reshape((len(images), n_vertices, n_points))
+    texture = np.nanmean(all_samples, axis=2)
+    return texture
+
+
+def niimg_to_surf_data(image, surf_mesh,
+                       radius=3., kind='line', interpolation='nearest',
+                       n_samples=None, mask=None):
+    """Extract surface data from a Nifti image.
+
+    Parameters
+    ----------
+
+    image : niimg-like object, 3d or 4d.
+
+    surf_mesh : str or numpy.ndarray
+        Either a file containing surface mesh geometry (valid formats
+        are .gii or Freesurfer specific files such as .orig, .pial,
+        .sphere, .white, .inflated) or a list of two Numpy arrays,
+        the first containing the x-y-z coordinates of the mesh
+        vertices, the second containing the indices (into coords)
+        of the mesh faces.
+
+    radius : float, optional (default=3.).
+        The size (in mm) of the neighbourhood from which samples are drawn
+        around each node.
+
+    kind : {'line', 'ball'}
+        The strategy used to sample image intensities around each vertex.
+
+        - 'line' (the default):
+            samples are regularly spaced along the normal to the mesh, over the
+            interval [-radius, +radius].
+            (sometimes called thickness sampling)
+        - 'ball':
+            samples are regularly spaced inside a ball centered at the mesh
+            vertex.
+
+    interpolation : {'linear', 'nearest'}
+        How the image intensity is measured at a sample point.
+
+        - 'nearest' (the default):
+            Use the intensity of the nearest voxel.
+        - 'linear':
+            Use a trilinear interpolation of neighbouring voxels.
+
+    n_samples : int or None, optional (default=None)
+        How many samples are drawn around each vertex and averaged. If None,
+        use a reasonable default for the chosen sampling strategy ('ball' or
+        'line').
+
+    mask : niimg-like object or None, optional (default=None)
+        Samples falling out of this mask or out of the image are ignored.
+        If None, don't apply any mask.
+
+    Returns
+    -------
+    texture: array, 1d or 2d.
+        If image was a 3d image (e.g a stat map), a 1d vector is returned,
+        containing one value for each mesh node.
+        If image was a 4d image, a 2d array is returned, where each row
+        corresponds to a mesh node.
+
+    """
+    image = nilearn.image.load_img(image)
+    if mask is not None:
+        mask = resampling.resample_to_img(
+            mask, image, interpolation='nearest').get_data()
```

Comment:
Is it a good idea to use the same interpolation method as specified/specifies at the function level ?

Reply:
This is not related to sampling around mesh vertices, it is just to align the mask and the image. I use nearest because it is fast and preserves the datatype of the mask (continuous would transform the boolean mask into a floating-point image)